### PR TITLE
Deploy Gateway DB to Railway

### PR DESCRIPTION
**Pull request by pikonha**
- **chore: replace yarn by npm**
- **chore: build ccip-server**
- **chore: build gateway**
- **feat: gateway on docker**
- **chore: start from bundle file**
- **feat: add railway config**
- **fix: remove minify from gateway build command**
- **chore: use GATEWAY_URL env to deploy the contracts**
- **refactor: railway config gateway:db**
- **chore: update README with npm and deploy commands**
- **fix: working dev and start gateway:db**
- **feat(husky hooks): commitilint with npm and commitizen**
- **fix: contracts lint**

# Deploy Gateway DB to Railway

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

The Gateway can be reached through [this url](https://db-gateway-production.up.railway.app/) which was created on my Railway's personal account.

* Dockerizes the Gateway alongside its private dependencies
* Replace yarn with npm due to building limitation
* Use `commitizen` for conventional commits

## Related Issue

#80 

## Changes
- [ ] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [ ] Documentation update
- [X] Other: infra for deployment

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: packages/contracts/script/local/DatabaseResolver.s.sol [hunk 1]
```diff
@@ -11,13 +11,14 @@ import {DatabaseResolver} from "../../src/DatabaseResolver.sol";
...
     function run() external {
+        string memory gatewayURL = vm.envString("GATEWAY_URL");
         uint256 privateKey = vm.envUint("PRIVATE_KEY");
         address publicKey = vm.addr(privateKey);
         vm.startBroadcast(privateKey);
 
         ENSRegistry registry = new ENSRegistry();
         string[] memory urls = new string[](1);
-        urls[0] = "http://localhost:3000/{sender}/{data}.json";
+        urls[0] = gatewayURL;
```
Comment (LeonardoVieira1630) [inline packages/contracts/script/local/DatabaseResolver.s.sol:21]: Nice upgrade.

File: packages/gateway/package.json [hunk 2]
```diff
@@ -2,19 +2,23 @@
...
+  "types": "dist/database.mjs",
   "scripts": {
     "test": "vitest run",
     "test:watch": "vitest",
-    "start:db": "dotenvx run --env-file=../../.env -- ts-node ./cmd/database.ts",
-    "start:arb": "dotenvx run --env-file=../../.env -- ts-node ./cmd/arbitrum.ts",
-    "dev:db": "ENV_FILE=../../.env nodemon ./cmd/database.ts",
-    "dev:arb": "ENV_FILE=../../.env nodemon ./cmd/arbitrum.ts",
-    "lint": "eslint . --ext .ts --fix"
+    "start:db": "node ./dist/database.mjs",
```
Comment (LeonardoVieira1630) [inline packages/gateway/package.json:10]: Isnt necessary to run the dotenvx command before it anymore?

**Review by LeonardoVieira1630 — COMMENTED** [review]
Great Pull Request Lucas. I had just a few points to clarify.

File: packages/gateway/package.json (hunk 2 above)
Comment (pikonha) [reply packages/gateway/package.json:10]: this is for running on the cloud so it expects the environment variables to already be present

**Pull request merged**